kernel/shared/src/main/scala/cats/effect/kernel/syntax/AsyncSyntax.scala

```diff
@@ -30,6 +32,9 @@ final class AsyncOps[F[_], A] private[syntax] (private[syntax] val wrapped: F[A]
   def evalOn(ec: ExecutionContext)(implicit F: Async[F]): F[A] =
     Async[F].evalOn(wrapped, ec)
 
+  def evalOn(executor: Executor)(implicit F: Async[F]): F[A] =
+    Async[F].evalOnExecutor(wrapped, executor)
```

Comment:
```suggestion
  def evalOnExecutor(executor: Executor)(implicit F: Async[F]): F[A] =
    Async[F].evalOnExecutor(wrapped, executor)
```

Would you mind also adding this method directly on `IO`? Thanks!

Reply:
@armanbilge hello! Could you please check?)